Multisite Scheduled Updates: Add global navigation Plugins menu item

bogiii opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Closes https://github.com/Automattic/wp-calypso/issues/90577

## Proposed Changes

* Creates plugins section specific global navigation menu items

## Testing Instructions

* Go to `/sites`
* Check if there is a "Plugins" menu item
* Click on it and check if there is a "Scheduled updates" feature renders
<img width="611" alt="Screenshot 2024-05-16 at 18 23 32" src="https://github.com/Automattic/wp-calypso/assets/1241413/4120a979-f3e2-46da-ab78-fa38f0d80057">

![Screen Capture on 2024-05-16 at 11-38-35](https://github.com/Automattic/wp-calypso/assets/1241413/b3d33e12-75a2-4578-944d-c5b0284a01c6)

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

bogiii committed:
Multisite Scheduled Updates: Add global navigation Plugins menu item

bogiii committed:
Update comment

lucasmendes-design commented:
Also added Jana for the review here :)

lucasmendes-design commented:
Hi there, great to see this coming to life. Some notes: 

- Honestly, I think it's strange to have the Scheduled Updates without the Plugins list. I know this is on purpose but I just want to highlight. 

- We need to change the icon here. 
<img width="490" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/dc6af09a-b981-4223-a8aa-b91d675c8e88">
<img width="566" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/0c6b44e3-5b53-4470-bb67-c623fd72dcd1">

- The collapsed sidebar has a strange shape. It should be a square. 
<img width="737" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/4421deaf-67f9-4ab1-a7b0-b0bb37c705cc">
<img width="1196" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/f8412829-82bb-4d89-b3f9-8d55537bedc5">

- Different header
<img width="1723" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/cb95ca30-01d8-4f8a-81d0-42ae45b53323">
<img width="1717" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/0768584a-108f-4d43-a885-ab604db90892">
<img width="1722" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/f4a5d611-cc64-47ae-a533-53624a4a0ea3">


- The Back button is wrong here. 
<img width="708" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/28a58fe0-e067-45ba-b438-334af8fb56c7">
<img width="910" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/bd5d176c-4fc7-497e-ad48-763363f30ef5">

- Calendar icon is strange. Which one is this? 
<img width="321" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/06531388-3e4a-4375-8678-ff5dc76cec1c">

cc: @JanaMW27

lucasmendes-design commented:
Search field: After you type, the icon should be an "X"
<img width="522" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/5eb8fde0-f841-4962-88d6-50e1e07d35ec">
<img width="178" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/cd660ae0-8c07-49d9-a1e6-f2b79a278e51">

lucasmendes-design commented:
How can I test the complete flow, or is this PR just to check if it's rendered? 

<img width="1052" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/957a5cc4-f77a-4820-98c9-0541e1219f51">

obenland commented:
I think this PR is just to integrate the feature into the multisite navigation.

To test the whole flow, create a WoA site and [install a plugin from the Marketplace](https://wordpress.com/plugins?s=email+notification).

lucasmendes-design commented:
@davemart-in in case you want to have a look at this :)

lucasmendes-design commented:
> To test the whole flow, you'll want to create a WoA site and [install a plugin from the Marketplace](https://wordpress.com/plugins?s=email+notification).

I tried, but it doesn't show any plugin to schedule, and I also chose one that is not made by us. I'll try it again, thanks.

lucasmendes-design commented:
I know this is not the place but I'm not sure where to share this. I'm ccing @JanaMW27 :) 

More food for thought: 
-  I think we should display a counter to inform users of the number of sites selected. 
- The search field is working awkwardly. 
- I would add "select all" sites option. Not sure if this is relevant

https://github.com/Automattic/wp-calypso/assets/98343583/7179cf2e-abf6-48ad-ad9e-eea37e953202

bogiii commented:
I would like to mention that this PR only changes the global navigation menu. The rest has already been merged but hidden from the others because of the missing entry point, which was added by this PR. We can wait to merge this one until we are completely ready.


> We need to change the icon here.

I have followed the way we use icons on other menu items. It's a dashicons pallet, and I just picked the plugin icon that already exists.

I'll pull SVG icon from the Figma and replace it.

> The collapsed sidebar has a strange shape. It should be a square.
> Calendar icon is strange. Which one is this?

I did not change or override any style there; I configured the menu item following the defined rules. The calendar icon is also from the Dashicons pallet.

> Different header

We consume the same component in different contexts.
You are right for the "Scheduled Updates" title; in the multisite context, it needs to be 20px instead of 24px, which is correct for the single-site context.
The "New schedule" title has correct style following: https://www.figma.com/design/rrEvrq63njWdmuXkJSTcY2/Plugin-scheduler---multisite?node-id=3081-88566&t=4c8atqclI7Wi14UC-0

> The Back button is wrong here. It's also not "< Sites" it should be Plugins.

I thought that "< Plugins" was a mistake, and I replaced it with "< Sites" because when you click on that menu item, we are going back to the Sites and not Plugins. 🤔 

Hmmmm, how did you get this screen?
<img width="910" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/bd5d176c-4fc7-497e-ad48-763363f30ef5">

> Search field: After you type, the icon should be an "X"

Here we use SearchControl from the `@wordpress/components` package https://wordpress.github.io/gutenberg/?path=/docs/components-searchcontrol--docs
There must be some missing configuration. I'll fix it. 

@lucasmendes-design, thanks for the feedback.

For further discussion, I suggest not posting the "Multisite Scheduled Updates" feature feedback here in the PR. Let's use our team's P2 space to make it easier to follow.

bogiii committed:
Update plugins icon

bogiii committed:
Update calendar icon

lucasmendes-design commented:
Sorry @bogiii. Next time, I'll post on the Slack channel. Jana just shared with me the correct one :)

lucasmendes-design commented:
> Hmmmm, how did you get this screen?

It's here -> https://agencies.automattic.com/ but this is the same pattern Core uses. 
<img width="509" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/237b23e3-26e9-435c-abce-ef2bc3e65f15">

> Icons 

Calendar: https://www.figma.com/design/eqlepL7MHifK8Bgr1OWqLr?node-id=6927-26213#804265808
Plugin: https://www.figma.com/design/eqlepL7MHifK8Bgr1OWqLr?node-id=6927-26869#804266699 

> I did not change or override any style there; I configured the menu item following the defined rules. 

So it's a bug =/ the way we're displaying the collapsable menu is strange. If you go to [/horizon](https://horizon.wordpress.com/) we have the right one
![image](https://github.com/Automattic/wp-calypso/assets/98343583/89a9e7a7-ff38-4c9a-bd94-3ac74b24ee6d)
<img width="933" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/3c33da37-2d1e-4da4-9ad3-ed41445def1f">

Thanks 🙏🏽

bogiii committed:
Create Plugins sidebar component

bogiii committed:
Remove wrong way of adding plugins sidebar menu

bogiii commented:
Hey @lucasmendes-design,

I've created a Plugins sidebar menu following the existing solutions you shared above.
Also, I've updated the icons and fixed the size issues when the menu is collapsed.

Please take another look when you find some time.

lucasmendes-design commented:
Hi @bogiii, it's looking perfect :) 

This is not a blocker by ay means but maybe we can come up with some copy below the title here. I believe the plan is to add the plugin list in the future, so we can have a nice copy saying that "plugins list soon ...." and/or something about the "Scheduled Updates". WDYT? 

<img width="869" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/53a3835f-1ace-4f3b-ac0e-afedf4883b5b">
<img width="729" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/d757c3b9-ead6-495e-8ec9-4fa16d133cbf">

bogiii commented:
Absolutely, I agree.

I would rather provide info about the "Scheduled Updates" feature than the Plugins in general since it's currently one menu item.

My proposal:
<img width="498" alt="Screenshot 2024-05-16 at 14 42 15" src="https://github.com/Automattic/wp-calypso/assets/1241413/8dbb7b0e-98f6-4489-824b-6f9e94643dd2">

What do you think? @obenland

michaelpick commented:
Feel free to [drop a request our way](https://dotcomcopyrequests.wordpress.com/2024/04/10/hello-world/) any time.

TimBroddin commented:
Just wondering. The padding of the menu item changes when opening details. Is this intentional?

![CleanShot 2024-05-16 at 15 13 47@2x](https://github.com/Automattic/wp-calypso/assets/528287/af736b7b-8ffc-4dd3-b8ef-a2bad573cf7c)
![CleanShot 2024-05-16 at 15 14 00@2x](https://github.com/Automattic/wp-calypso/assets/528287/d6ecffab-ba04-4a07-b255-395a959b5261)

lucasmendes-design commented:
Not sure if I can reproduce this but it should be the same as the main navigation. 

<img width="659" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/d6f847e7-7814-4465-be3b-4fa119570a68">

obenland commented:
> I would reduce the copy and make it smaller.

In i1 we had a subheading of "Streamline your workflow with scheduled updates, timed to suit your needs." Let's reuse that.

bogiii committed:
Add plugins menu description

bogiii commented:
> Just wondering. The padding of the menu item changes when opening details. Is this intentional?

It's not intentional. I can not find a pattern when it happens, but it has happened once to me even without these changes in this PR.
Since it's not introduced by this PR, I'll leave the fix for future PR.

The final change looks:
<img width="611" alt="Screenshot 2024-05-16 at 18 23 32" src="https://github.com/Automattic/wp-calypso/assets/1241413/4120a979-f3e2-46da-ab78-fa38f0d80057">

bogiii committed:
Hide Plugins menu item under the feature flag

bogiii committed:
Replace feature flag

ouikhuan reviewed (approved):
Apart from what's discussed above, everything else looks good to me 👍

lucasmendes-design commented:
💯

tyxla commented on client/my-sites/plugins/sidebar/index.tsx:
```diff
@@ -0,0 +1,42 @@
+		<GlobalSidebar
+			className={ clsx( 'sidebar--plugins', { 'is-collapsed': isCollapsed } ) }
+			siteTitle={ ! isCollapsed && translate( 'Plugins' ) }
+			requireBackLink={ true }
```
Please, rebase your PRs before merging to trunk. Because #90499 landed before merging this PR, this line was introducing an ESLint error. I've resolved this in #90822.

bogiii replied:
Sorry to hear that, but I actually rebased it a few hours before merging back to the trunk.

tyxla replied:
Curious. Something might be off with the check, I'll investigate.

tyxla replied:
Fix coming up in https://github.com/Automattic/wp-calypso/pull/90877. Sorry for the noise!
